### PR TITLE
Various docs tasks for consistency

### LINKED ISSUE #270: Make MCMC docs more consistent with GP and EKP docs

### LINKED ISSUE #274: Write "Recommended configuration for GP/RF" at top of docs pages and improve defaults

### PR DESCRIPTION
**Pull request by odunbar**
## Purpose 

Closes #270
Closes #274

## Content

- Adds new Sinusoid example images and link into landing page, reorganize landing page.
[Landing page](https://clima.github.io/CalibrateEmulateSample.jl/previews/PR271/)

- Rearrange docs pages order (for cleaner contents)
- Rename `"Package-design/API for AbstractMCMC"` -> `"Sample"`
- Adds a user interface section into the `"Sample"` page at the top. Bumping other information further down
[Sample page](https://clima.github.io/CalibrateEmulateSample.jl/previews/PR271/sample/)

- Adds recommended constructor for RF (nonseparable + low rank)
[RF page](https://clima.github.io/CalibrateEmulateSample.jl/previews/PR271/random_feature_emulator/)

## Also
- Updates testing to Julia v1.10, as pkg downloader would not work on 1.7.3 thus we could not merge PRs

----
- [ ] I have read and checked the items on the review checklist.

### CODE REVIEW

File: docs/src/index.md [hunk 1]
```diff
@@ -25,25 +25,46 @@ Given an observation ``y``, the computer model ``\mathcal{G}``, the observationa
...
-### The three steps of the algorithm:
+### The three steps of the algorithm: see our walkthrough of the [Sinusoid Example](@ref)
 
-The **calibrate** step of the algorithm consists of an application of [Ensemble Kalman Processes](https://github.com/CliMA/EnsembleKalmanProcesses.jl), which generates input-output pairs ``\{\theta, \mathcal{G}(\theta)\}`` in high density around an optimal parameter ``\theta^*``. This ``\theta^*`` will be near a mode of the posterior distribution (Note: This the only time we interface with the forward model ``\mathcal{G}``).
+**Given some noisy observations...**
+```@raw html
+<img src="assets/sinusoid_true_vs_observed_signal.png" width="400">
+```
+
+1. The **calibrate** step of the algorithm consists of an application of [Ensemble Kalman Processes](https://github.com/CliMA/EnsembleKalmanProcesses.jl), which generates input-output pairs ``\{\theta, \mathcal{G}(\theta)\}`` in high density around an optimal parameter ``\theta^*``. This ``\theta^*`` will be near a mode of the posterior distribution (Note: This the only time we interface with the forward model ``\mathcal{G}``).
```
Comment (lm2612) [inline docs/src/index.md:35]: typo - This is the only time

File: docs/src/index.md [hunk 2]
```diff
@@ -25,25 +25,46 @@ Given an observation ``y``, the computer model ``\mathcal{G}``, the observationa
...
+
+**calibrate with EKP to generate data pairs...**
+```@raw html
+<img src="assets/sinusoid_eki_pairs.png" width="400">
+```
 
-The **emulate** step takes these pairs ``\{\theta, \mathcal{G}(\theta)\}`` and trains a statistical surrogate model (e.g., a Gaussian process), emulating the forward map ``\mathcal{G}``.
+2. The **emulate** step takes these pairs ``\{\theta, \mathcal{G}(\theta)\}`` and trains a statistical surrogate model (e.g., a Gaussian process), emulating the forward map ``\mathcal{G}``.
```
Comment (lm2612) [inline docs/src/index.md:42]: in the built docs the 2. and 3. appear as 1s

File: docs/src/sample.md [hunk 3]
```diff
@@ -1,14 +1,66 @@
...
+One can quickly plot the marginals of the prior and posterior distribution with
+```julia
+using Plots
+plot(prior)
+plot!(posterior)
+```
+or extract statistics of the (unconstrained) distribution with
+```julia
+mean_posterior = mean(posterior)
+cov_postierior = cov(posterior)
```
Comment (lm2612) [inline docs/src/sample.md:60]: cov_postierior -> cov_posterior

**Review by lm2612 — COMMENTED** [review]
Looks really good, the side menu and sample page are much cleaner! only a few minor typos

File: docs/src/index.md (hunk 2 above)
Comment (odunbar) [reply docs/src/index.md]: Removed numbering, as it seems markdown won't do custom bullets

**Comment by odunbar** [comment]
Julia-wide registry issues are causing Buildkite to fail (whereas I can rerun test suites), I will bypass the buildkite to merge as we have not changed any of the files being run.

**Pull request merged**